myia/compile/backends/pytorch.py

```diff
@@ -484,13 +505,22 @@ def from_backend_value(self, v, t):
         if isinstance(t, abstract.AbstractScalar):
             return self.to_scalar(v)
         elif isinstance(t, abstract.AbstractArray):
-            return self.to_numpy(v)
+            # Convert torch tensor to numpy tensor.
+            output = self.to_numpy(v)
+            # If possible and necessary, cast numpy tensor to expected tensor.
+            array_type = t.element.xtype()
```

Comment:
If you ever need to do that, then whatever function produced the array has a bug.  All primitives should return the right type.

Reply:
@abergeron It seems we cannot cast values to neither `uint16` nor `uint32` nor `uint64` (unsigned integers except `uint8`) in Pytorch. This code currently fails:
```python
import numpy as np
from myia import myia
from myia.operations import array_cast


@myia(backend='pytorch')
def f(val):
    return array_cast(val, np.uint32)


def main():
    val = np.arange(4, dtype='float32')
    res = f(val)
    print(res.dtype)


if __name__ == '__main__':
    main()
```

With this error:
```
  File "/home/notoraptor/mila/dev/git/myia/myia/compile/backends/pytorch.py", line 178, in pytorch_array_cast
    dt = _type_map[t.value.xtype()]

KeyError: UInt[32]
```

Indeed, `_type_map` does not handle unsigned types very well: https://github.com/mila-iqia/myia/blob/master/myia/compile/backends/pytorch.py#L16

Checking torch documentation, it seems indeed some torch dtypes are missing: https://pytorch.org/docs/stable/tensor_attributes.html#torch-dtype

Reply:
Ok then for this change, but this is just more motivation for me to drop the pytorch backend.